### PR TITLE
Code cleanup on battle related scripts

### PR DESCRIPTION
**Pull request by kori4r2**

### CODE REVIEW

File: Assets/Scripts/Battle/BattleManager.cs [hunk 1]
```diff
@@ -1,264 +1,252 @@
...
-                if (debugAction.action.triggered) {
-                    foreach (BattleUnit bUnit in battleUnits) {
-                        if (bUnit.CurHP > 0) {
-                            bUnit.DecreaseHP(1.0f);
-                            UpdateLives();
-                        }
-                    }
-                }
+            if (!StaticReferences.DebugBuild || !debugAction.action.triggered)
+                return;
```
Comment (sortudo) [inline Assets/Scripts/Battle/BattleManager.cs:54]: 🚀

File: Assets/Scripts/Battle/BattleManager.cs [hunk 2]
```diff
@@ -1,264 +1,252 @@
...
+            StartFirstTurn();
         }
 
-        public void ShowEnemyInfo() {
-            enemyContent.ShowEnemyInfo(currentUnit);
+        public void StartFirstTurn() {
+            CurrentUnit = queue.Dequeue();
+            if (CurrentUnit != null)
+                CurrentUnit.OnTurnStart?.Invoke(CurrentUnit);
         }
```
Comment (sortudo) [inline Assets/Scripts/Battle/BattleManager.cs]: StartFirstTurn is very similar to StartNextTurn, can't we just put them together?

File: Assets/Scripts/Battle/BattleSkillManager.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            CurrentUser = unit;
+        }
+
+        public static void SelectSkill(Skill skillSelected) {
+            CurrentSkill = skillSelected;
+        }
+
+        public static TargetType GetSkillType() {
+            return CurrentSkill != null ? CurrentSkill.target : TargetType.Null;
+        }
```
Comment (sortudo) [inline Assets/Scripts/Battle/BattleSkillManager.cs]: GetSkillType actually get target type?

File: Assets/Scripts/Battle/SkillVFX.cs [hunk 4]
```diff
@@ -1,41 +1,52 @@
...
         }
 
-        public void SetTarget(BattleUnit unit, bool isCallback = false) {
-            forceCallback = isCallback;
-            GetComponent<SpriteRenderer>().sortingOrder = unit.GetComponent<SpriteRenderer>().sortingOrder + 2;
+        public void SetTarget(BattleUnit unit) {
+            isCallback = false;
+            spriteRenderer.sortingOrder = unit.GetComponent<SpriteRenderer>().sortingOrder + 2;
+            UpdateLocalPosition(unit);
+        }
```
Comment (sortudo) [inline Assets/Scripts/Battle/SkillVFX.cs]: Why did you split SetTarget like this?

File: Assets/Scripts/Battle/SkillVFX.cs [hunk 5]
```diff
@@ -56,57 +67,44 @@ public enum TargetPosition {
...
             particleList.Add(particle);
             ParticleSystemRenderer renderer = particle.GetComponent<ParticleSystemRenderer>();
             if (!renderer)
                 return;
-            renderer.sortingLayerID = GetComponent<SpriteRenderer>().sortingLayerID;
-            renderer.sortingLayerName = GetComponent<SpriteRenderer>().sortingLayerName;
-            renderer.sortingOrder = GetComponent<SpriteRenderer>().sortingOrder;
+            renderer.sortingLayerID = spriteRenderer.sortingLayerID;
+            renderer.sortingLayerName = spriteRenderer.sortingLayerName;
+            renderer.sortingOrder = spriteRenderer.sortingOrder;
```
Comment (sortudo) [inline Assets/Scripts/Battle/SkillVFX.cs:102]: 🚀

File: Assets/Scripts/Battle/StatusEffect.cs [hunk 6]
```diff
@@ -14,58 +14,62 @@ namespace FinalInferno {
...
+            return Target.effects.Find(effect => effect.GetType() == searchedType) == this;
+        }
+
+        protected bool TargetHasEffectOfType(System.Type searchedType) {
+            return Target.effects.Find(effect => effect.GetType() == searchedType) != null;
+        }
+
+        protected bool SourceHasEffectOfType(System.Type searchedType) {
+            return Source.effects.Find(effect => effect.GetType() == searchedType) != null;
+        }
```
Comment (sortudo) [inline Assets/Scripts/Battle/StatusEffect.cs:60]: Can't this 3 methos just be one?

File: Assets/Scripts/Battle/StatusVFXHandler.cs [hunk 7]
```diff
@@ -147,140 +12,127 @@ private class IndividualHandler {
...
-
             int index = (int)effect.VFXID;
             if (handlers[index] == null || handlers[index].effects.Contains(effect)) {
-                // Se o handler não existir é porque o efeito visual não foi implementado
-                // Se o efeito em questão ja está sendo mostrado ele não deve ser adicionado de novo
-                if (handlers[index] != null) {
-                    Debug.LogError("Tentou adicionar um efeito que ja estava no handler");
-                }
+                if (handlers[index] != null)
+                    Debug.LogError($"Tentou adicionar o efeito {effect} que ja estava no handler", this);
```
Comment (sortudo) [inline Assets/Scripts/Battle/StatusVFXHandler.cs]: LogError in Portuguese

**Review by sortudo — APPROVED** [review]
Nice job! ⭐

File: Assets/Scripts/Battle/StatusEffect.cs (hunk 6 above)
Comment (kori4r2) [reply Assets/Scripts/Battle/StatusEffect.cs:60]: These methods were created to make code of child classes more legible, merging them together would require finding an appropriate generic name and adding at least two more parameters to the methods, which defeats that purpose

**Pull request merged**